test/lib/usd/translators/testUsdImportCustomConverter.py

```diff
@@ -0,0 +1,182 @@
+#!/pxrpythonsubst
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pxr import Usd
+from pxr import UsdShade
+
+import json
+import os
+import unittest
+
+from maya import cmds
+from maya import standalone
+
+import fixturesUtils
+
+class testUsdImportCustomConverter(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.input_path = fixturesUtils.setUpClass(__file__)
+
+        cls.test_dir = os.path.join(cls.input_path,
+                                    "UsdImportCustomConverterTest")
+
+        cls.usd_path = os.path.join(cls.test_dir,
+                                    'UsdImportCustomConverterTest.usda')
+
+        cmds.workspace(cls.test_dir, o=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        standalone.uninitialize()
+
+    def setUp(self):
+        """Clear the scene"""
+        cmds.file(f=True, new=True)
+        self.default_materials = cmds.ls(materials=True)
+
+    def checkMaterials(self, expected):
+        """Check that all the cubes are linked to the right material"""
+        for cube_name, material_type in expected:
+            sgs = cmds.listConnections(cube_name, type="shadingEngine")
+            shaders = cmds.listConnections(sgs[0] + ".surfaceShader")
+            if material_type:
+                node_type = cmds.nodeType(shaders[0])
+                self.assertEqual(node_type, material_type)
+            else:
+                self.assertTrue(shaders[0] in self.default_materials)
+
+    def testImportCommand(self):
+        """
+        Tests a custom exporter for a conversion that exists in an unloaded
+        plugin using the mayaUSDImport command.
+        """
+        modes = [["useRegistry", "maya"],
+                 ["useRegistry", "UsdPreviewSurface"],
+                 ["displayColor", "default"]]
```

Comment:
Is `default` significant here, or is it just ignored?

Reply:
It is ignored for "displayColor" shading mode importer.
I do not like passing an unused argument, but I feel this is only transitional: in a future where we keep only the "useRegistry" importer, we can remove the first argument and use only a list of materialConversion instead of a list of {mode, conversion}.